graphene-reader/src/main/kotlin/com/graphene/reader/store/key/elasticsearch/handler/TagBasedKeySearchHandler.kt

```diff
@@ -40,11 +40,16 @@ class TagBasedKeySearchHandler(
 
       while (response.hits.hits.isNotEmpty()) {
         for (hit in response.hits) {
-          val path = Path(hit.id)
+          val sources = hit.sourceAsMap as Map<String, *>
+          if (! sources.containsKey(KEY)) {
+            continue
+          }
+          val path = Path(sources[KEY].toString())
           if (Objects.nonNull(hit.sourceAsMap)) {
-            val tags = (hit.sourceAsMap as Map<String, *>)
-            for (tag in tags) {
-              path.addTag(tag.key, tag.value.toString())
+            for (source in sources) {
+              if (! INTERNAL_KEYS.contains(source.key)) {
```

Comment:
It might be a stupid question. :)
When can this happen?

Reply:
It always happens because indexed keys are in the form: 
```
{
 "@key": "cpu_idle;host=a", 
 "@name":"cpu_idle",
 "@tags": ["host"], 
 "host":"a"
}
```
It is okay to put those values into Path() but excluded because I think it is useless.

Reply:
I simply modified the logic for that and added the test code:
https://github.com/graphene-monitoring/graphene/pull/82/commits/2ddd93e29f497010f94dcd340ca427be7cda0bcc